test/use-recurly.test.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import { suppressConsoleErrors } from './support/helpers';
+
+import {
+  RecurlyProvider,
+  Elements,
+  useRecurly
+} from '../lib';
+
+import { RecurlyElementsContext } from '../lib/elements';
+
+describe('useRecurly', function () {
+  describe('when not a descendant of <Elements />', function () {
+    const message = /you are trying to use Recurly outside of an Elements context/;
+
+    suppressConsoleErrors();
+
+    it('throws an error', function () {
+      expect(() => {
+        render(<TestComponent />);
+      }).toThrow(message);
+
+      expect(() => {
+        render(
+          <RecurlyProvider publicKey="test-public-key">
+            <TestComponent />
+          </RecurlyProvider>
+        );
+      }).toThrow(message);
+    });
+
+    function TestComponent () {
+      const recurly = useRecurly();
+      return '';
+    }
+  });
+
+  describe('when a descendant of <Elements />', function () {
+    const fixture = TestComponent => () => {
+      render(
+        <RecurlyProvider publicKey="test-public-key">
+          <Elements>
+            <TestComponent />
+          </Elements>
+        </RecurlyProvider>
+      );
+    };
+
+    it('returns a Recurly instance', function () {
+      expect(fixture(TestComponent)).not.toThrow();
+
+      function TestComponent () {
+        const recurly = useRecurly();
+        expect(recurly.token).toBeInstanceOf(Function);
+        expect(recurly.config.publicKey).toBe('test-public-key');
+        return '';
+      }
+    });
+
+    describe('recurly.token', function () {
+      it('tokenizes the parent Elements instance', function () {
+        fixture(TestComponent)();
+
+        function TestComponent () {
+          const recurly = useRecurly();
+          const elements = React.useContext(RecurlyElementsContext).elements;
+          const underlyingRecurly = elements.recurly;
```

Comment:
The extraction of the underlying `Recurly` instance here relies on knowing that an `Elements` instance carries a reference to the `Recurly` that created it. While that's outside of the domain of this library, I think it's a reasonable access point in order to ensure that the `token` method is being called with the expected bound `Elements` instance.